File: backend/src/main/java/me/stefan923/waterly/entity/ScheduleSettings.java
package me.stefan923.waterly.entity;

import lombok.*;

import java.time.DayOfWeek;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class ScheduleSettings {

    private TimeRange sunday = new TimeRange();
    private TimeRange monday = new TimeRange();
    private TimeRange tuesday = new TimeRange();
    private TimeRange wednesday = new TimeRange();
    private TimeRange thursday = new TimeRange();
    private TimeRange friday = new TimeRange();
    private TimeRange saturday = new TimeRange();

    public TimeRange getTimeRangeByDayOfWeek(DayOfWeek dayOfWeek) {
        return switch (dayOfWeek) {
            case SUNDAY -> sunday;
            case MONDAY -> monday;
            case TUESDAY -> tuesday;
            case WEDNESDAY -> wednesday;
            case THURSDAY -> thursday;
            case FRIDAY -> friday;
            case SATURDAY -> saturday;
        };
    }

}

File: frontend/Waterly/src/entity/ScheduleSettings.swift
//
//  ScheduleSettings.swift
//  Waterly
//
//  Created by Stefan Popescu on 12.06.2023.
//

import Foundation

struct ScheduleSettings: Codable {
    var monday: TimeRange
    var tuesday: TimeRange
    var wednesday: TimeRange
    var thursday: TimeRange
    var friday: TimeRange
    var saturday: TimeRange
    var sunday: TimeRange

    init(monday: TimeRange, tuesday: TimeRange, wednesday: TimeRange, thursday: TimeRange, friday: TimeRange, saturday: TimeRange, sunday: TimeRange) {
        self.sunday = sunday
        self.monday = monday
        self.tuesday = tuesday
        self.wednesday = wednesday
        self.thursday = thursday
        self.friday = friday
        self.saturday = saturday
    }
    
    private enum CodingKeys: String, CodingKey {
        case monday
        case tuesday
        case wednesday
        case thursday
        case friday
        case saturday
        case sunday
    }
}
